docs/src/main/asciidoc/_elytron/Elytron_WildFly_Java_Security_Manager.adoc

```diff
@@ -0,0 +1,327 @@
+
+The Java Enterprise Edition (EE) 7 specification introduced a new feature which allows application developers to specify a Java Security Manager (JSM) policy for their Java EE applications, when deployed to a compliant Java EE Application Server such as WildFly. Until now, writing JSM policies has been pretty tedious. Now a new tool has been developed which allows the generation of a JSM policy for deployments running on WildFly. It is possible that running with JSM enabled may affect performance, https://bugs.openjdk.java.net/browse/JDK-8043631[JEP 232] indicates the performance impact would be 10-15%, but it is still recommended to test the impact per application.
+
+Elytron WildFly Java Security Manager basic enhancements are
+
+* Support of JavaEE 7 (and newer) https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/permissions_7.xsd[Java EE application permissions schema] to allow declarative security per application deployment.
+* Support of minimum and maximum permissions per server instance.
+* Support of a "log-only" mode to ease development and testing of permissions.
+* Output of debug information to detect missing permissions.
+* Output of trace information to analyse special situations with {oracle-javadoc}/java.base/java/security/ProtectionDomain.html[protection domains], classloaders, and {oracle-javadoc}/java.base/java/security/CodeSource.html[codesources].
+
+==== Why Run with the Java Security Manager Enabled?
+Running a JSM will not fully protect the server from malicious attackers exploiting security vulnerabilities. It does, however, offer another layer of protection which can help reduce the impact of serious security vulnerabilities, such as deserialization attacks. For example, most of the recent https://www.github.com/mbechler/marshalsec/blob/master/marshalsec.pdf?raw=true[attacks against Jackson Databind] rely on making a Socket connection to an attacker-controlled JNDI Server to load malicious code. https://access.redhat.com/solutions/3279231[This article] provides information on how this issue potentially affects an application written for JBoss EAP, which would be same for WildFly. The Security Manager could block the socket creation, and potentially thwart the attack.
+
+==== JavaSE 17 Deprecate the Security Manager for Removal
+Unfortunately the community decided to deprecate the Java Security Manager with Java 17. The full details are described in https://openjdk.org/jeps/411[JEP 411: Deprecate the Security Manager for Removal]. WildFly {wildflyversion} still tests and actively supports the Security Manager. Used in conjunction with other tools such as [Serialization Filtering](https://docs.oracle.com/en/java/javase/11/core/serialization-filtering1.html#GUID-3ECB288D-E5BD-4412-892F-E9BB11D4C98A) the JSM is still a good defense in depth measure.
```

Comment:
Just a minor, I would not use the word `unfortunately` in documentation as it is just an opinion and would depend on the context? And the links do not render well here